### PR TITLE
feat: Text constructor, overflow, and softWrap docs

### LINKED ISSUE #63311: TextOverflow not working as expected in Text widget
## Steps to reproduce

Run the following app (also visible at http://dartpad.dev/0c6794bd5daf972aff91724efee85dd9):

```dart
import 'package:flutter/widgets.dart';

void main() {
  runApp(BugReport());
}

class BugReport extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Directionality(
      textDirection: TextDirection.ltr,
      child: Align(
        alignment: Alignment.center,
        child: SizedBox(
          width: 150,
          child: ColoredBox(
            color: const Color(0xff666666),
            child: Text('onelongword anotherreallylongword', maxLines: 1, overflow: TextOverflow.visible),
          ),
        ),
      ),
    );
  }
}
```

## Expected behavior

You expect something like the following result:

bug_desired png

## Actual behavior

This is what's rendered -- the second word doesn't show up at all, despite the overflow being set to `visible`:

bug_actual

## Related

If you change the overflow to `TextOverflow.clip`, then you expect the following output:

bug_desired_clip

But again, the second word is completely omitted:

bug_actual_clip

## Other info

In the latter case, if clipping at word breaks is actually working as intended, then I think we should add a new `TextOverflow` enum value that allows the caller to clip in the middle of a word.  The first case above seems unlikely that it's working as intended.

### PR DESCRIPTION
**Pull request by broccolism**
## Description

Add documentation about overflow and softWrap property of Text widget, which means that the softWrap property affects the overflow behavior.

## Related Issues

fixes [#63311](https://github.com/flutter/flutter/issues/63311)

## Tests

N/A

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text.dart [hunk 1]
```diff
@@ -126,9 +126,14 @@ class DefaultTextStyle extends InheritedTheme {
   /// Whether the text should break at soft line breaks.
   ///
   /// If false, the glyphs in the text will be positioned as if there was unlimited horizontal space.
+  /// 
+  /// This also decides basis of [overflow] property.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/text.dart:130]: This should say how it influences the overflow property.

File: packages/flutter/lib/src/widgets/text.dart [hunk 2]
```diff
@@ -126,9 +126,14 @@ class DefaultTextStyle extends InheritedTheme {
...
   final bool softWrap;
 
   /// How visual overflow should be handled.
+  /// 
+  /// This will be applied on a word basis when [softWrap] is true, and on an alphabet basis when
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/text.dart:135]: What is "word base" or "alphabet base"? I think we'd need to explain this a little better.

File: packages/flutter/lib/src/widgets/text.dart [hunk 3]
```diff
@@ -341,6 +346,9 @@ class Text extends StatelessWidget {
   /// closest enclosing [DefaultTextStyle].
   ///
   /// The [data] parameter must not be null.
+  /// 
+  /// The [overflow] property will be applied on a word basis when [softWrap] is true,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/text.dart:350]: same here.

**Review by goderbauer — COMMENTED** [review]
Thanks for the contribution. Can you please also fix the whitespace errors that the analyzer check is complaining about? Thanks.

File: packages/flutter/lib/src/widgets/text.dart (hunk 3 above)
Comment (broccolism) [reply packages/flutter/lib/src/widgets/text.dart]: Sorry for late response. I tried to fix, please check them out!

File: packages/flutter/lib/src/widgets/text.dart [hunk 4]
```diff
@@ -126,9 +126,15 @@ class DefaultTextStyle extends InheritedTheme {
   /// Whether the text should break at soft line breaks.
   ///
   /// If false, the glyphs in the text will be positioned as if there was unlimited horizontal space.
+  ///
+  /// This also decides the [overflow] property's behavior. If this is true or null,
+  /// the word causing overflow will not be shown at all.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/text.dart:131]: Here and below
```suggestion
  /// the glyph causing overflow, and those that follow, will not be rendered.
```

**Review by Piinks — COMMENTED** [review]
Hey @broccolism welcome! It looks like the analyzer is failing here, can you update to resolve? :)

**Comment by broccolism** [comment]
Okay! I commited change.

**Comment by broccolism** [comment]
Oh, I'm sorry. I'll update and commit again ASAP.

**Comment by broccolism** [comment]
@Piinks I fixed some expressions and tried to resolve problems. Could you check this out? thanks.

**Review by Piinks — APPROVED** [review]
LGTM! Thank you for improving our documentation! 🎉

**Pull request merged**